plumbing/msg/waiter.go

```diff
@@ -109,7 +110,11 @@ func (w *Waiter) findMessage(ctx context.Context, ts types.TipSet, msgCid cid.Ci
 		}
 		for i := 0; i < iterator.Value().Len(); i++ {
 			blk := iterator.Value().At(i)
-			for _, msg := range blk.Messages {
+			msgs, err := w.messageReader.LoadMessages(ctx, blk.Messages)
```

Comment:
Does this get replaced with a graphsync query? If so, it would be good to get a TODO and an issue about that.

Reply:
No this is unrelated to the graphsync query.

Reply:
I thought the point of graphsync was to allow us to retrieve only the data we need. In this case it seems wasteful to load all the block messages when we only care about one message and we know its CID. It is not possible to use graphsync here?

Reply:
There must have been a communication gap on the purpose of graphsync.  The point of graphsync is to fetch lots of ipld nodes over the network in one request.  This code is a local search of the blockchain data that has already been fetched.  There are other efforts to merkelize the message collection which could lead to better targeted searches in this code.  However these efforts are orthogonal.  Message collections are currently just cbor arrays of messages and not setup for any sort of smart searching.  The only way to find a message is to follow the hash link and search all of the messages underneath it.